tuned/plugins/plugin_bootloader.py

```diff
@@ -355,20 +357,46 @@ def _cmdline(self, enabling, value, verify, ignore_missing):
 				return None
 			cmdline_set = set(cmdline.split())
 			value_set = set(v.split())
-			cmdline_intersect = cmdline_set.intersection(value_set)
-			if cmdline_intersect == value_set:
+			missing_set = value_set - cmdline_set
+			if len(missing_set) == 0:
 				log.info(consts.STR_VERIFY_PROFILE_VALUE_OK % ("cmdline", str(value_set)))
 				return True
 			else:
-				log.error(consts.STR_VERIFY_PROFILE_VALUE_FAIL % ("cmdline", str(cmdline_intersect), str(value_set)))
+				cmdline_dict = {v.split("=", 1)[0]: v for v in cmdline_set}
+				for m in missing_set:
+					arg = m.split("=", 1)[0]
+					if not arg in cmdline_dict:
+						log.error(consts.STR_VERIFY_PROFILE_VALUE_FAIL % ("cmdline arg "+arg, "<missing>", m))
```

Comment:
Please output arg value in quotes to be consistent with other logging, i.e.:
`cmdline arg 'ARGUMENT'` instead of `cmdline arg ARGUMENT`

Reply:
I added the quotes, but I'm not sure the result is what you intended... it now has nested quotes:
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: 'cmdline arg 'intel_pstate'' = '<missing>', expected 'intel_pstate=disable'
```
Examples from other plugins:
```
DEBUG    tuned.plugins.plugin_scheduler: Read SMP affinity of IRQ '39': '[0, 1, 2]'
ERROR    tuned.plugins.plugin_scheduler: verify: failed: 'SMP affinity of IRQ 39' = '[0, 1, 2]', expected '[0, 1, 3]'
[...]
DEBUG    tuned.plugins.plugin_sysctl: Value of sysctl parameter 'kernel.timer_migration' is '0'
INFO     tuned.plugins.base: verify: passed: 'kernel.timer_migration' = ''
```

We could simply leave out the "cmdline arg" and make it:
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: 'intel_pstate' = '<missing>', expected 'intel_pstate=disable'
```

Reply:
You are right, please drop the nested quotes, the following could work, but I have problem that it's not consistent, i.e.  `'intel_pstate' = '<missing>'` vs `'intel_pstate=disable'`:
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: cmdline arg 'intel_pstate' = '<missing>', expected 'intel_pstate=disable'
```
Could you rewrite it with the one level quotes and consistent output, e.g.:
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: cmdline arg 'intel_pstate' = '<missing>', expected 'intel_pstate' = 'disable'
```
or:
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: cmdline arg 'intel_pstate = <missing>', expected 'intel_pstate = disable'
```
Or similarly?

Reply:
The trouble comes with using the format string `consts.STR_VERIFY_PROFILE_VALUE_FAIL`, which is a bit limiting here.

If it is ok that I don't use it, then I would like to distinguish the two cases "missing" and "set to something else", and make the output:
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: cmdline arg 'intel_pstate' is missing, expected 'intel_pstate=disable'
```
and
```
ERROR    tuned.plugins.plugin_bootloader: verify: failed: cmdline arg 'intel_pstate=something_else', expected 'intel_pstate=disable'
```

Reply:
Could you use the `consts.STR_VERIFY_PROFILE_VALUE_FAIL` for the case where it's different and introduce new const, e.g. `consts.STR_VERIFY_PROFILE_VALUE_FAIL_MISSING` for the missing case?

Reply:
The existing `STR_VERIFY_PROFILE_VALUE_FAIL` still seems a bit awkward (it would give `'cmdline arg intel_pstate' = 'something_else'`, and I think it would be better to show the string exactly as found in the cmdline). So I'd rather introduce two new format strings (`STR_VERIFY_PROFILE_CMDLINE_FAIL` and `STR_VERIFY_PROFILE_CMDLINE_FAIL_MISSING`?) to produce the messages as shown in my previous comment.

Reply:
I pushed an update with two new format strings.